orc8r/cloud/go/obsidian/swagger/specs.go

```diff
@@ -0,0 +1,87 @@
+/*
+ Copyright 2021 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+// Specs provide access to swagger spec YAML files.
+type Specs interface {
+	GetCommonSpec() (string, error)
+	GetPartialSpec(service string) (string, error)
+	GetStandaloneSpec(service string) (string, error)
+}
+
+// fsSpecs implements Specs via the filesystem at a specific base dir.
+type fsSpecs struct {
+	baseDir string
+}
+
+var _ Specs = (*fsSpecs)(nil)
+
+const commonSpecRelPath = "common/swagger-common.yml"
+
+// GetCommonSpec returns ./common/swagger-common.yml
+func (f *fsSpecs) GetCommonSpec() (string, error) {
+	return readEntireFileString(filepath.Join(f.baseDir, commonSpecRelPath))
+}
+
+// GetPartialSpec returns ./partial/$service.swagger.v1.yml
+func (f *fsSpecs) GetPartialSpec(service string) (string, error) {
+	return readEntireFileString(partial.path(f.baseDir, service))
+}
+
+// GetStandaloneSpec returns ./standalone/$service.swagger.v1.yml
+func (f *fsSpecs) GetStandaloneSpec(service string) (string, error) {
+	return readEntireFileString(standalone.path(f.baseDir, service))
+}
+
+// NewFSSpecs returns Specs backed by fsSpecs using the specified base path.
+// See also GetDefaultSpecs which embeds the default base path and is preferred
+// when appropriate.
+func NewFSSpecs(baseDir string) Specs {
+	return &fsSpecs{baseDir: baseDir}
+}
+
+const defaultSpecBasePath = "/etc/magma/swagger/specs"
+
+// GetDefaultSpecs returns Specs with the default production base path.
+func GetDefaultSpecs() Specs {
+	return NewFSSpecs(defaultSpecBasePath)
+}
+
+type specType string
+
+const (
+	partial    specType = "partial"
+	standalone specType = "standalone"
+)
+
+func (t specType) path(base string, service string) string {
+	return filepath.Join(base, string(t), fmt.Sprintf("%s.swagger.v1.yml", strings.ToLower(service)))
+}
+
+// readEntireFileString wraps ioutil.ReadFile and returns a string iff the
+// entire file can be read without error.
+func readEntireFileString(path string) (string, error) {
```

Comment:
Overall it's a nit, this filename is fine, I just prefer to have overly simple+concise names to get the gist, then use docstring for higher-resolution understanding